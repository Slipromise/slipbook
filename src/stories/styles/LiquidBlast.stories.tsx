import type { Meta, StoryObj } from "@storybook/react";
import React from "react";
import style from "./liquidBlast.module.scss";

type Props = {
  circleClassName?: React.HtmlHTMLAttributes<HTMLDivElement>["className"];
};

const LiquidBlast = ({ circleClassName }: Props) => {
  return (
    <div className={style.container}>
      <div className={circleClassName}>
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
        <i className={style.particle} />
      </div>
      <svg className={style.filterSvg}>
        <filter id="gooey">
          <feGaussianBlur in="SourceGraphic" stdDeviation="10" />
          <feColorMatrix
            values="1 0 0 0 0
          0 1 0 0 0
          0 0 1 0 0
          0 0 0 50 -10"
          />
        </filter>
      </svg>
    </div>
  );
};
const meta: Meta<typeof LiquidBlast> = {
  component: LiquidBlast,
  parameters: {
    layout: "centered",
  },
};

export default meta;

type Story = StoryObj<typeof LiquidBlast>;

export const Normal: Story = {
  args: {
    circleClassName: style.normal,
  },
};

export const Slime: Story = {
  args: {
    circleClassName: style.slime,
  },
};
