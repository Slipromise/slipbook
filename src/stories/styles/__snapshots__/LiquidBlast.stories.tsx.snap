// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stories/styles/LiquidBlast Normal smoke-test 1`] = `
<div class="_container_15o0e_11">
  <div class="_normal_15o0e_14">
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
  </div>
  <svg class="_filterSvg_15o0e_459">
    <filter id="gooey">
      <feGaussianBlur in="SourceGraphic"
                      stddeviation="10"
      >
      </feGaussianBlur>
      <feColorMatrix values="1 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 50 -10">
      </feColorMatrix>
    </filter>
  </svg>
</div>
`;

exports[`stories/styles/LiquidBlast Slime smoke-test 1`] = `
<div class="_container_15o0e_11">
  <div class="_slime_15o0e_193">
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
    <i class="_particle_15o0e_24">
    </i>
  </div>
  <svg class="_filterSvg_15o0e_459">
    <filter id="gooey">
      <feGaussianBlur in="SourceGraphic"
                      stddeviation="10"
      >
      </feGaussianBlur>
      <feColorMatrix values="1 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 50 -10">
      </feColorMatrix>
    </filter>
  </svg>
</div>
`;
